include/picongpu/plugins/PhaseSpace/PhaseSpaceFunctors.hpp

```diff
@@ -209,19 +203,23 @@ namespace picongpu
 
             FunctorParticle<r_dir, num_pbins, SuperCellSize> functorParticle;
 
-            particleAccess::Cell2Particle<SuperCellSize, numWorkers> forEachParticleInCell;
-            forEachParticleInCell(
-                acc,
-                /* mandatory params */
-                particlesBox,
+            auto superCellIdx = indexBlockOffset / static_cast<pmacc::math::Int<simDim>>(SuperCellSize::toRT());
```

Comment:
Is the guard size handled correctly here (i mean, incides with guard vs. without)? I was not able to quickly understand it.

Reply:
Yes the handling is correct and equal to the way it was handled before the refactoring: https://github.com/ComputationalRadiationPhysics/picongpu/pull/4289/files#diff-436b5c29b52b01c3a106e7c1e6c7b3b23befa57765b56b01a7be30e6e7509644L58